scaleway/resource_baremetal_server.go

```diff
@@ -272,6 +314,68 @@ func resourceScalewayBaremetalServerUpdate(ctx context.Context, d *schema.Resour
 		return diag.FromErr(err)
 	}
 
+	server, err := baremetalAPI.GetServer(&baremetal.GetServerRequest{
+		Zone:     zonedID.Zone,
+		ServerID: zonedID.ID,
+	})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	var serverGetOptionIDs []*baremetal.ServerOption
+	for i := range server.Options {
+		serverGetOptionIDs = append(serverGetOptionIDs, server.Options[i])
+	}
+
+	if d.HasChange("options") {
+		options, err := expandBaremetalOptions(d.Get("options"))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		if len(options) > 0 {
+			optionsToDelete := baremetalCompareOptionIDsToDelete(options, serverGetOptionIDs, zonedID.Zone)
+			for i := range optionsToDelete {
+				_, err = baremetalAPI.DeleteOptionServer(&baremetal.DeleteOptionServerRequest{
+					Zone:     server.Zone,
+					ServerID: server.ID,
+					OptionID: optionsToDelete[i].ID,
+				})
+				if err != nil {
+					return diag.FromErr(err)
+				}
+			}
+
+			_, err = waitForBaremetalServerOptions(ctx, baremetalAPI, zonedID.Zone, zonedID.ID, d.Timeout(schema.TimeoutDelete))
+			if err != nil && !is404Error(err) {
+				return diag.FromErr(err)
+			}
+
+			optionsToAdd := baremetalCompareOptionIDsToAdd(options, serverGetOptionIDs, zonedID.Zone)
+			for i := range optionsToAdd {
+				_, err = baremetalAPI.AddOptionServer(&baremetal.AddOptionServerRequest{
+					Zone:      server.Zone,
+					ServerID:  server.ID,
+					OptionID:  optionsToAdd[i].ID,
+					ExpiresAt: optionsToAdd[i].ExpiresAt,
+				})
+				if err != nil {
+					return diag.FromErr(err)
+				}
+			}
+		} else {
+			for i := range serverGetOptionIDs {
+				_, err = baremetalAPI.DeleteOptionServer(&baremetal.DeleteOptionServerRequest{
```

Comment:
Why is there twice a DeleteOptionServerRequest? Could it be simplified in one loop?